lib/lumberjack/server.rb

```diff
@@ -219,40 +221,54 @@ def initialize(fd)
 
     def run(&block)
       while true
-        # TODO(sissel): Ack on idle.
-        # X: - if any unacked, IO.select
-        # X:   - on timeout, ack all.
-        # X: Doing so will prevent slow streams from retransmitting
-        # X: too many events after errors.
-        @parser.feed(@fd.sysread(16384)) do |event, *args|
-          case event
-          when :window_size; window_size(*args, &block)
-          when :data; data(*args, &block)
-          end
-          #send(event, *args)
-        end # feed
-      end # while true
+        read_socket(&block)
+      end
     rescue EOFError, OpenSSL::SSL::SSLError, IOError, Errno::ECONNRESET
+
       # EOF or other read errors, only action is to shutdown which we'll do in
       # 'ensure'
     ensure
       close rescue 'Already closed stream'
     end # def run
 
+    def read_socket(&block)
+      # TODO(sissel): Ack on idle.
+      # X: - if any unacked, IO.select
+      # X:   - on timeout, ack all.
+      # X: Doing so will prevent slow streams from retransmitting
+      # X: too many events after errors.
+      @parser.feed(@fd.sysread(READ_SIZE)) do |event, *args|
+        case event
+        when :window_size; window_size(*args, &block)
```

Comment:
style - both `when` should be either in-line or on separate lines. Maybe remove the `;` and move the `window_size` call down a line?